workbench-core/accounts/src/utilities/hostingAccountLifecycleService.test.ts

```diff
@@ -600,4 +626,113 @@ describe('HostingAccountLifecycleService', () => {
+    const sampleBucketArn = `arn:aws:s3:::${sampleBucketName}`;
+    const sampleAccountId = '123456789012';
+
+    // Mock S3 calls
+    const s3Mock = mockClient(S3Client);
+    s3Mock.on(PutBucketPolicyCommand).resolves({});
+    const basePolicy = PolicyDocument.fromJson(
+      JSON.parse(`{
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Sid": "Deny requests that do not use SigV4",
+                "Effect": "Deny",
+                "Principal": {
+                    "AWS": "*"
+                },
+                "Action": "s3:*",
+                "Resource": "${sampleBucketArn}/*",
+                "Condition": {
+                    "StringNotEquals": {
+                        "s3:signatureversion": "AWS4-HMAC-SHA256"
+                    }
+                }
+            }
+        ]
+    }`)
+    );
+
+    const expectedPolicy = PolicyDocument.fromJson(
+      JSON.parse(`
+      {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Sid": "Deny requests that do not use SigV4",
+                "Effect": "Deny",
+                "Principal": {
+                    "AWS": "*"
+                },
+                "Action": "s3:*",
+                "Resource": "${sampleBucketArn}/*",
+                "Condition": {
+                    "StringNotEquals": {
+                        "s3:signatureversion": "AWS4-HMAC-SHA256"
+                    }
+                }
+            },
+            {
+                "Sid": "List:environment-files",
+                "Effect": "Allow",
+                "Principal": {
+                    "AWS": "arn:aws:iam::${sampleAccountId}:root"
+                },
+                "Action": "s3:ListBucket",
+                "Resource": "${sampleBucketArn}",
+                "Condition": {
+                    "StringLike": {
+                        "s3:prefix": "environment-files*"
+                    }
+                }
+            },
+            {
+                "Sid": "Get:environment-files",
+                "Effect": "Allow",
+                "Principal": {
+                    "AWS": "arn:aws:iam::${sampleAccountId}:root"
+                },
+                "Action": "s3:GetObject",
+                "Resource": "${sampleBucketArn}/environment-files*"
+            }
+        ]
+    }`)
+    );
+
+    const postUpdatedPolicy = hostingAccountLifecycleService['_updateBucketPolicyDocumentWithAllStatements'](
+      sampleBucketArn,
+      sampleAccountId,
+      basePolicy
+    );
+    expect(postUpdatedPolicy).toEqual(expectedPolicy);
+  });
+
+  test('buildTemplateURLForAccount basic unit test', async () => {
+    const service = new HostingAccountLifecycleService();
+    const sampleExternalId = 'sample';
+    const region = process.env.AWS_REGION!;
+
+    service['_aws'].clients.s3 = new S3({
+      region : region,
+      credentials: {
+        accessKeyId: "AKIAIOSFODNN7EXAMPLE",
```

Comment:
Do we still need this mock? If yes, can we just have both value be `EXAMPLE` instead of a realistic looking key value in the prefix?